app/javascript/react/screens/App/Analytics/screens/AnalyticsSummary/AnalyticsSummary.js

```diff
@@ -53,7 +74,48 @@ class AnalyticsSummary extends React.Component {
     }
   }
 
+  fetchProviders = () => this.props.fetchProvidersAction(VMWARE_PROVIDERS_FILTERS, PROVIDER_REFRESH_ATTRIBUTES);
+
+  doInitialFetch = () => {
+    const { fetchReportsAction } = this.props;
+    this.fetchProviders();
+    fetchReportsAction(VM_SUMMARY_REPORT_FILTERS);
+    fetchReportsAction(ENV_SUMMARY_REPORT_FILTERS);
+  };
+
+  retry = () => {
+    this.props.resetAllStateAction();
+    this.doInitialFetch();
+  };
+
+  handleUpdateForProviders = prevProps => {
+    const { isFetchingProviders } = this.props;
+
+    // If we fetched providers, and there are no refresh errors, and some are awaiting refresh, wait and fetch again.
+    if (
+      prevProps.isFetchingProviders &&
+      !isFetchingProviders &&
+      someProvidersExist(this.props) &&
+      noRefreshErrorsExist(this.props) &&
+      someProvidersAwaitingRefresh(this.props)
+    ) {
+      this.fetchProvidersTimeout = setTimeout(this.fetchProviders, 3000);
```

Comment:
Also here: why do we need the timeout?

Reply:
This one is for polling. This helper method is called in componentDidUpdate, and this condition will run if a fetch for providers just completed but some were still waiting for a refresh to complete. This will fetch them again every 3 seconds until they are confirmed to be refreshed.